main.js

```diff
@@ -124,20 +159,21 @@ if (config.downloadJAD) {
   loadingMIDletPromises.push(new Promise(function(resolve, reject) {
     JARStore.loadJAR("midlet.jar").then(function(loaded) {
       if (loaded) {
+        if (!document.hidden) {
+          // Show the splash screen as soon as possible.
+          showSplashScreen();
```

Comment:
I guess there's no way to show this earlier, because we don't know if we need to download the MIDlet or not.